octue/resources/datafile.py

```diff
@@ -31,6 +31,7 @@
 logger = logging.getLogger(__name__)
 
 
+downloaded_files = []
```

Comment:
Make this a dictionary and use analysis IDs as keys to avoid potential thread safety issues

Reply:
I've decided against this and instead included a warning and a way to switch off deletion if thread safety is needed